ggshield/core/git_shell.py

```diff
@@ -202,6 +216,55 @@ def get_staged_filepaths(wd: Optional[str] = None) -> List[Path]:
     return [Path(path_str) for path_str in filepaths]
 
 
+def get_diff_files_status(
+    wd: Optional[str], ref: str, staged: bool = False, similarity: int = 100
+) -> Dict[Path, Filemode]:
+    """
+    Fecthes the statuses of modified files since a given ref.
+    """
+
+    def parse_name_status_patch(patch: str) -> Dict[Path, Filemode]:
+
+        m = {
+            "A": Filemode.NEW,
+            "D": Filemode.DELETE,
+            "M": Filemode.MODIFY,
+            "T": Filemode.MODIFY,
```

Comment:
"M" and "T" are both modified is this wanted?

Reply:
Yes, T is a type change, like symbolic link to hard link. In practice, it does not appear that much.

Reply:
Nit: add a link to the documentation.